Feature/sortable monthly stats

Veith3n opened the pull request.

Veith3n committed:
add dotenv and rack-cors

Veith3n committed:
create env.example

Veith3n committed:
add configuration for rack-cors

Veith3n committed:
move data serialization logic to presenter

Veith3n committed:
update factory for trips

Veith3n committed:
add tests for trip_presenter and update tests for trip_service

Veith3n committed:
update trips_controller

-allow call to this controller without authenticity token
- add two params in monthly method (orderParam and orderType)
- returns JSONs with different codes in different cases

Veith3n committed:
update tests for trips_controller

Veith3n committed:
update tests for trip

Veith3n committed:
update README

Veith3n committed:
add reek

Veith3n committed:
add monthly_grouped_by_date scope in trip.rb

Veith3n committed:
add trip_repository_mock

Veith3n committed:
code refactor

Veith3n committed:
update tests

Veith3n committed:
update trip_controller_test

ulska commented on app/controllers/trips_controller.rb:
```diff
@@ -1,21 +1,53 @@
 
   def create
-    @trip = Trip.create(trip_params)
+    @trip = Trip.new(trip_params)
```
Are you sure you need instance variable here?

ulska commented on app/models/trip.rb:
```diff
@@ -5,6 +5,7 @@ class Trip < ApplicationRecord
 
   scope :weekly, -> { where(date: Date.current.beginning_of_week..Date.current.end_of_week).order(:date) }
   scope :monthly, -> { where(date: Date.current.beginning_of_month..Date.current.end_of_month).order(:date) }
+  scope :monthly_grouped_by_date, -> { monthly.group_by(&:date) }
```
Scopes shouldnt be responsible for grouping.

ulska commented on spec/factories/trips.rb:
```diff
@@ -1,7 +1,7 @@
   factory :trip do
-    sequence(:start_address) { |n| "Miodowa #{n}, Warszawa" }
-    sequence(:destination_address) { |n| "Nowogordzka #{n}, Warszawa" }
+    start_address 'Miodowa 1, Warszawa'
```
:+1:

ulska commented on app/controllers/trips_controller.rb:
```diff
@@ -1,21 +1,53 @@
-    render json: TripStats.monthly_stats.to_json
+    if order_param_correct? && order_type_correct?
+      orderParam = sort_params[:orderParam]
+      orderType = sort_params[:orderType]
```
Is it neccessary to extract `sort_params[:orderParam]` and `sort_params[:orderType]` to separate variables?

aarek replied:
Use `snake_case` instead of `camelCase` for variables names ;)

ulska commented on spec/controllers/trips_controller_spec.rb:
```diff
@@ -16,4 +16,20 @@
+
+      get :monthly, params: params
+
+      expect(response).to have_http_status(422)
```
You should also check the error message, not only status.

Veith3n committed:
check error message in test

Veith3n committed:
update gems versions

Veith3n committed:
refactor trips factory

Veith3n committed:
make geocoder raise exception for OverQueryLimitError and Timeout error

Veith3n committed:
update vcr cassette call for trip_controller_spec

Veith3n committed:
code refactor

- remove unnecessary instance  variable in trips_controller
- move scope into method in trip.rb
